### DISCUSSION TITLE
Configurations via Dart file instead of Yaml

### DISCUSSION THREAD

**Discussion by mannprerak2** [serverpod/serverpod#186] [General]
Just throwing this out there. Wouldn't Dart files be a much better choice for configurations?

I mean they can provide auto-complete suggestions, and its easy to know what config does what because documentation is just a click away (or hover)

Take Django for example, all the models/serializers/settings are defined in python files.

**Comment by vlidholt** [comment]
That may not be a bad idea. Will consider this as a future update!

**Reply by anandsubbu007** [reply]
May be like isar db
With the help of build runner.

**Comment by vlidholt** [comment]
I've looked into this a bit more and it may be hard to do. The configuration files aren't just used when starting the server, but also by the tooling. E.g. the command-line tool and the visual interface (these are both still being developed).

**Reply by mannprerak2** [reply]
I agree, I guess when Dart will introduce Meta Programming this might be a good thing to revisit.
